### DISCUSSION TITLE
[Question] How import a module (bluetooth support with bleak)?

### DISCUSSION THREAD

**Discussion by 9acca9** [flet-dev/flet#4154] [Q&A]
### Question

So, i compile with this sentence:
p4a create --requirements bleak --arch arm64-v8a --sdk-dir $ANDROID_SDK_ROOT --ndk-dir $ANDROID_SDK_ROOT/ndk/$NDK_VERSION --dist-name mydist

If i try to compile again with the same sentence, i get:
```
mydist: min API 21, includes recipes (hostpython3, libffi, openssl, sqlite3, python3, genericndkbuild, setuptools, six, pyjnius, android, bleak), built for archs (arm64-v8a)
[INFO]:    mydist has compatible recipes, using this one

```

So, yep, it seems that bleak is included.

later i make:
flet build apk
(and that leave me a big big apk of 109 MB)

Anyway i intall in my cellphone and im getting this module not found... named "bleak".

So, what im doing wrong??
thanks!

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by FeodorFitsner** [comment]
I'd post this question in Kivy repo.

**Reply by 9acca9** [reply]
why? i mean. bleak is not just a package as any other? (im using it to communicate with a BLE device)

**Reply by FeodorFitsner** [reply]
I will look how it could be built with the latest Flet pre-release which does not use kivy.

re: kivy command - you can ask there why it's not building.

**Reply by 9acca9** [reply]
oh, ok. But i think must be something with Flet because i was using it with Kivy and was working fine.

The thing is, that i remember now...
How you can import a module but not from PyPi?
Because you have to specify this to use the "develop" branch of Bleak and not the stable.
`P4A_bleak_DIR="path to develop bleak source" buildozer android debug` (of course, that is with buildozer)

It is possible to import that kind of module? (i mean, not from pypi)
Thanks!

**Comment by FeodorFitsner** [comment]
Looks like bleak is specifically made for Kivy: https://pypi.org/project/bleak/#files
https://github.com/hbldh/bleak/tree/develop/bleak/backends

I tried to build it with mobile forge and it was not recognized as a binary module.

**Reply by FeodorFitsner** [reply]
I guess we still need to implement bluetooth support in Flet with something from: https://pub.dev/packages?q=bluetooth&sort=top

**Reply by 9acca9** [reply]
oh, ok. thanks for give it a try!
Sorry the ignorance but it is possible to implement that? (https://pub.dev/packages?q=bluetooth&sort=top) (your link) or must be... "ported"? 
if i cant use it then i will try to create the app with Kivy, im  a beginner, so, sadly, i cant help to create or recreate those elements.
Thanks for the try and the answers!

**Comment by 2692596549** [comment]
Hello, have you solved your problem? I'm having the same issue.
